README.md
# Calc_Billing

history.txt
History Total Bayar Pengguna :


main.py
from tkinter import *
import tkinter as tk
from tkinter import filedialog
from PIL import ImageTk,Image
import os

root = Tk()
root.title('π warnet')

root.state('normal')
root.resizable(False, False)

my_img1 = ImageTk.PhotoImage(Image.open("Image/1.jpg"))
my_img2 = ImageTk.PhotoImage(Image.open("Image/2.jpg"))
my_img3 = ImageTk.PhotoImage(Image.open("Image/3.jpg"))
my_img4 = ImageTk.PhotoImage(Image.open("Image/4.jpg"))
my_img5 = ImageTk.PhotoImage(Image.open("Image/5.jpg"))

image_list = [my_img1, my_img2, my_img3, my_img4, my_img5]

my_label = Label(image=my_img1)
my_label.grid(row=0, column=0, columnspan=5)

def forward(image_number):
    global my_label
    global button_forward
    global button_back

    my_label.grid_forget()
    my_label = Label(image=image_list[image_number-1])
    button_forward = Button(root, text=">>", command=lambda: forward(image_number+1))
    button_back = Button(root, text="<<", command=lambda: back(image_number-1))

    if image_number == 5:
        button_forward = Button(root, text=">>", state=DISABLED)

    my_label.grid(row=0, column=0, columnspan=5)
    button_back.grid(row=1, column=0)
    button_forward.grid(row=1, column=4)
    
def back(image_number):
    global my_label
    global button_forward
    global button_back


    my_label.grid_forget()
    my_label = Label(image=image_list[image_number-1])
    button_forward = Button(root, text=">>", command=lambda: forward(image_number+1))
    button_back = Button(root, text="<<", command=lambda: back(image_number-1))


    if image_number == 1:
        button_back = Button(root, text="<<", state=DISABLED)
        

    my_label.grid(row=0, column=0, columnspan=5)
    button_back.grid(row=1, column=0)
    button_forward.grid(row=1, column=4)


def file_opener():
    #set to correct destination
    history = 'history.txt'
    os.system('"%s"' % history)

def about_opener():
    #set to correct destination
    about = 'about.txt'
    os.system('"%s"' % about)

def open_billing():
    #set to correct destination
    billing = 'billing.py'
    os.system('"%s"' % billing)
    
button_history = Button(root, text="π History", command= lambda:file_opener())
button_billing = Button(root, text="Billing Warnet", command= lambda:open_billing())
button_back = Button(root, text="<<", command=back, state=DISABLED)
button_forward = Button(root, text=">>", command=lambda: forward(2))
lbl_cp = Button(root, text="copyright @ π Team", command= lambda:about_opener())

button_history.grid(row=1, column=1)
button_billing.grid(row=1, column=3)
button_back.grid(row=1, column=0)
button_forward.grid(row=1, column=4)
lbl_cp.grid(row=1, column=2)



root.mainloop()


billing_GUI.py
from tkinter import *
from tkinter import messagebox

main_window = Tk()
main_window.title('Kalkulator BMI Sederhana')

main_window.state('normal')
main_window.resizable(False, False)


def reset():
    input_stop.delete(0, END)
    input_start.delete(0, END)
    label_result['text'] = 0


def open_alert_error(msg):
    """
    menampilkan pesan error
    """
    w_alert = Toplevel(main_window)
    w_alert.title("Error")

    lbl_msg = Label(w_alert, text=msg)
    lbl_msg.grid(row=0, column=0)

    btn_ok = Button(w_alert, text="Ok", command=w_alert.destroy)
    btn_ok.grid(row=1, column=1)


def validate_input():
    """
    validasi semua input
    """
    status = True
    if input_start.get() == "":
        status = False
        open_alert_error("waktu mulai masih kosong")
    elif input_stop.get() == "":
        status = False
        open_alert_error("waktu selesai masih kosong")

    return status

# hitung


def hitung():

    if validate_input():
        end = float(input_stop.get())
        mulai = float(input_start.get())

        ab = (end-mulai)
        cd = 10*(end-mulai)
        ef = round(((ab*100)-40))
        gh = round(((ab*100)-40)/10)
        i = int(ab*3000)
        j = (cd/10)*3000
        k = abs(gh*500)
        l = round(cd % 10*500)
        if ((ab % 2) == 0)or(ab == 2):

            label_result['text'] = f"{i}"
        elif((cd % 10) == 0)and(cd >= 0):

            label_result['text'] = f"{j}"
        elif(ab < 0.10):
            label_result['text'] = f"{500}"
        elif(ab % 2) != 0:

            label_result['text'] = f"{k}"
        elif(end < mulai):
            label_result['text'] = f"{0}"
        else:

            label_result['text'] = f"{l}"


# left
judul = Label(main_window, text='Billing Warnet',
              font='arial 20 bold underline')
desc = Label(main_window, text='Harga per 10 menit adalah Rp.500')
start = Label(main_window, text='waktu mulai (j.m)')
stop = Label(main_window, text='waktu selesai')
input_start = Entry(main_window, width=25)
input_stop = Entry(main_window, width=25)
reset = Button(main_window, text='Reset', padx=40, command=reset)
submit = Button(main_window, text='Submit', padx=40, command=hitung)


judul.grid(row=0, columnspan=4)
desc.grid(row=1, column=0, sticky='w')
start.grid(row=3, column=0, sticky='w')
stop.grid(row=5, column=0, sticky='w')
input_start.grid(row=4, column=0, columnspan=2, sticky='w')
input_stop.grid(row=6, column=0, sticky='w')
reset.grid(row=7, column=0, sticky='w')
submit.grid(row=7, column=1, sticky='w')

# right
label_hitung = Label(main_window, text="Hasil Perhitungan", font=("Arial", 8))
label_result = Label(main_window, text="0", font=("Arial", 44))
lbl_cp = Label(main_window, text="Made in Indonesia")


label_hitung.grid(row=1, column=1)
label_result.grid(column=1, row=4, rowspan=2)
lbl_cp.grid(row=8, column=0)
main_window.mainloop()


billing.py
from tkinter import *
from tkinter import messagebox


main_window = Tk()
main_window.title('Billing Warnet')

main_window.state('normal')
main_window.resizable(False, False)

    
def reset():
    input_stop.delete(0,END)
    input_start.delete(0,END)
    label_result['text'] = 0
    f = open('history.txt', 'a')

#hitung
def hitung():
    f = open('history.txt', 'a')
    if input_start.get() == '':
        messagebox.showerror("Error", "waktu mulai tidak boleh kosong")
    else:
        if input_stop.get() == '':
            messagebox.showerror("Error", "waktu selesai tidak boleh kosong")            
        else:
            end = float(input_stop.get())
            mulai = float(input_start.get())
            ab=(end-mulai)
            cd=10*(end-mulai)
            ef=round(((ab*100)-40))
            gh=round(((ab*100)-40)/10)
            i=int(ab*3000)
            j=(cd/10)*3000
            k=abs(gh*500)
            l=round(cd%10*500)
            if ((ab % 2)==0)or(ab==2):
                z=int(ab*3000)
                f.write(str(z)+"\n")
                label_result['text'] = f"{i}"
                
            elif((cd % 10)==0)and(cd>=0):
                z=(cd/10)*3000
                f.write(str(z)+"\n")
                label_result['text'] = f"{j}"
                
            elif(ab<0.10):
                f.write(str("500")+"\n")
                f.close()
                label_result['text'] = f"{500}"
                
            elif(ab % 2)!=0:
                z=abs(gh*500)
                f.write(str(z)+"\n")
                label_result['text'] = f"{k}"
                
            elif(selesai<mulai):
                f.write(str("input salah")+"\n")
                label_result['text'] = f"{0}"
                
            else:
                z=round(cd%10*500)
                f.write(str(z)+"\n")
                label_result['text'] = f"{l}"
            f.close()   
    
#left
judul = Label(main_window, text='Billing Warnet',font='arial 20 bold underline')
desc = Label(main_window, text='Biaya per 10 menit adalah\nRp.500', pady=20, font='arial 10')
start = Label(main_window, text='waktu mulai (jj.mm)')
stop = Label(main_window, text='waktu selesai (jj.mm)')
input_start = Entry(main_window, width=25)
input_stop = Entry(main_window, width=25)
reset = Button(main_window, text='Reset', padx=40, command=reset)
submit = Button(main_window, text='Submit', padx=40, command=hitung)


judul.grid(row=0, columnspan=4 )
desc.grid(row=1, column=0, sticky='w')
start.grid(row=3, column=0,sticky='w')
stop.grid(row=5, column=0,sticky='w')
input_start.grid(row=4, column=0, columnspan=2, sticky='w')
input_stop.grid(row=6, column=0,sticky='w')
reset.grid(row=7, column=0,sticky='w')
submit.grid(row=7, column=1,sticky='w')

#right
label_hitung = Label(main_window, text="Total Bayar",font=("Arial bold",8))
label_result = Label(main_window, text="0",font=("Arial",44))
lbl_cp = Label(main_window, text="copyright @ π Team")


label_hitung.grid(row = 1, column=1)
label_result.grid(column=1, row=4, rowspan=2)
lbl_cp.grid(row=8, column=0, columnspan=2)



main_window.mainloop()

